modules/ide/xsk-ide-xsaccess/src/main/resources/META-INF/dirigible/xsk-ide-xsaccess/editor/editor.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+angular.module('page', [])
+  .controller('PageController', function ($scope) {
+
+    let messageHub = new FramesMessageHub();
+    let contents;
+
+    function getResource(thePath) {
+      let xhr = new XMLHttpRequest();
+      xhr.open('GET', thePath, false);
+      xhr.send();
+      if (xhr.status === 200) {
+        return xhr.responseText;
+      }
+    }
+
+    function loadContents(xsaccessfile) {
+      if (!xsaccessfile) {
+        console.error('XSACCESS file parameter is not present in the URL');
+      } else {
+        return getResource('/services/v4/ide/workspaces' + xsaccessfile);
+      }
+    }
+
+    function getViewParameters() {
+      if (window.frameElement.hasAttribute("data-parameters")) {
+        let readParams = JSON.parse(window.frameElement.getAttribute("data-parameters"));
+        $scope.file = readParams["file"];
+      } else {
+        let scanSearchParams = new URLSearchParams(window.location.search);
+        $scope.file = scanSearchParams.get('file');
+      }
+    }
+
+    $scope.initJSONcontent = function () {
+      if (!('allowMethods' in $scope.data.cors))
+        $scope.data.cors = {
+          ...$scope.data.cors,
+          allowMethods: $scope.allAllowedMethods,
+          allowOrigin: ["*", "*.sap.com"],
+          maxAge: 3600,
+          allowHeaders: [''],
+          exposeHeaders: ['']
+        };
+      if ($scope.data.headers.customHeaders && $scope.data.headers.customHeaders.length)
+        $scope.data.headers.customHeaders = $scope.data.headers.customHeaders.map((x) => {
+          return {
+            ...x,
+            thevalue: x.value.substr(0, 10) == 'ALLOW-FROM' ? 'ALLOW-FROM' : x.value,
+            url: x.value.substr(0, 10) == 'ALLOW-FROM' ? x.value.substr(11) : ''
+          }
+        })
+    }
+
+    function load() {
+      getViewParameters();
+      contents = loadContents($scope.file);
+      $scope.data = JSON.parse(contents);
+      $scope.anonymous_connection_allowed = $scope.data.anonymous_connection ? true : false;
+      $scope.allAllowedMethods = ["GET", "POST", "HEAD", "OPTIONS"];
```

Comment:
Why only those four methods?

Reply:
There are all allowed methods according to https://help.sap.com/docs/SAP_HANA_PLATFORM/400066065a1b46cf91df0ab436404ddc/a9fc5c220d744180850996e2f5d34d6c.html?version=1.0.12&locale=en-US (Table 9)

Reply:
This is not correct. The docs you mentioned say that those four http methods would be used by default, not that they are all that are allowed. What would happen if I need to use `PUT`? Or `PATCH`? Or `DELETE`?